session/session.go

```diff
@@ -825,12 +825,92 @@ func (s *session) SetGlobalSysVar(name, value string) error {
 	if err != nil {
 		return errors.Trace(err)
 	}
+
+	switch name {
+	case variable.TiDBMaxChunkSize:
+		return s.persistVarInConstraint(name, sVal, variable.TiDBInitChunkSize, func(s *session, val string) (string, error) {
+			newMaxChunkSize, err := strconv.Atoi(sVal)
+			if err != nil {
+				return "", errors.Trace(err)
+			}
+			initChunkSize, err := strconv.Atoi(val)
+			if err != nil {
+				return "", errors.Trace(err)
+			}
+			if initChunkSize > newMaxChunkSize {
+				return "", errors.Errorf("Try to set TiDB MaxChunkSize to %d small than InitChunkSize %d, please change InitChunkSize ahead",
+					newMaxChunkSize, initChunkSize)
+			}
+			return strconv.Itoa(newMaxChunkSize), nil
+		})
+	case variable.TiDBInitChunkSize:
+		return s.persistVarInConstraint(name, sVal, variable.TiDBMaxChunkSize, func(s *session, val string) (string, error) {
+			newInitChunkSize, err := strconv.Atoi(sVal)
+			if err != nil {
+				return "", errors.Trace(err)
+			}
+			maxChunkSize, err := strconv.Atoi(val)
+			if err != nil {
+				return "", errors.Trace(err)
+			}
+			if newInitChunkSize > maxChunkSize {
+				return "", errors.Errorf("Try to set TiDB InitChunkSize to %d big than MaxChunkSize %d, please change MaxChunkSize ahead",
+					newInitChunkSize, maxChunkSize)
+			}
+			return strconv.Itoa(newInitChunkSize), nil
+		})
+	default:
+		return s.persistVar(name, sVal)
+	}
+}
+
+func (s *session) persistVar(name, sVal string) error {
 	sql := fmt.Sprintf(`REPLACE %s.%s VALUES ('%s', '%s');`,
 		mysql.SystemDB, mysql.GlobalVariablesTable, strings.ToLower(name), sVal)
-	_, _, err = s.ExecRestrictedSQL(s, sql)
+	_, _, err := s.ExecRestrictedSQL(s, sql)
 	return errors.Trace(err)
 }
 
+func (s *session) persistVarInConstraint(setName, setVal, checkName string, condition func(s *session, val string) (string, error)) error {
+	retryCount := 11
```

Comment:
Why `retryCount` is 11?

Reply:
tidb auto retry use `10` in most situation 2 peoples change the same value in same time is unnormal, so retry 2-3 maybe can success.

by the way, I have delete those method in new commit- -  orz